### PR TITLE
Api changes for Voter adding option

### PR DESCRIPTION
**Pull request by JDawes-ScottLogic**
Refactored and renamed PollVoteController and TokenPollVoteController.
Refactored PollOptionController.

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Api/Controllers/PollOptionController.cs [hunk 1]
```diff
@@ -12,59 +11,23 @@ namespace VotingApplication.Web.Api.Controllers.API_Controllers
...
             {
                 if (optionCreationRequest == null)
                 {
                     ThrowError(HttpStatusCode.BadRequest);
                 }
 
-                Poll poll = context.Polls.Where(p => p.UUID == pollId).FirstOrDefault();
+                Poll poll = context
+                    .Polls
+                    .SingleOrDefault(p => p.UUID == pollId);
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Api/Controllers/PollOptionController.cs]: Should this have an Include(p => p.Options)?

File: VotingApplication/VotingApplication.Web/Api/Controllers/PollResultsController.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+                    if (poll.LastUpdated < clientLastUpdated)
+                    {
+                        return new HttpResponseMessage(HttpStatusCode.NotModified);
+                    }
+                }
+
+                List<Vote> votes = context
+                    .Votes
+                    .Include(v => v.Poll)
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Api/Controllers/PollResultsController.cs]: It doesn't look like we use vote.Poll anywhere

File: VotingApplication/VotingApplication.Web/App_Start/WebApiConfig.cs [hunk 3]
```diff
@@ -27,21 +27,21 @@ public static void Register(HttpConfiguration config)
             );
 
             config.Routes.MapHttpRoute(
-                name: "TokenPollVoteApiRoute",
-                routeTemplate: "api/token/{tokenGuid}/poll/{pollId}/vote/{voteId}",
-                defaults: new { controller = "TokenPollVote", voteId = RouteParameter.Optional }
+                name: "PollVoteApiRoute",
+                routeTemplate: "api/poll/{pollId}/token/{tokenGuid}/vote/{voteId}",
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/App_Start/WebApiConfig.cs]: The voteId is no longer used

File: VotingApplication/VotingApplication.Web/Api/Controllers/PollOptionController.cs (hunk 1 above)
Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web/Api/Controllers/PollOptionController.cs]: It seems to be working without it, but I'll add it in to make it clear.

File: VotingApplication/VotingApplication.Web/App_Start/WebApiConfig.cs (hunk 3 above)
Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web/App_Start/WebApiConfig.cs]: Removed.

**Comment by tpkelly** [comment]
(Y)

**Comment by stevenhillcox** [comment]
I am fine to pull this when it is mergable

**Pull request merged**